### DISCUSSION TITLE
Function/computed field type

### DISCUSSION THREAD

**Discussion by GP-S** [twentyhq/twenty#4117] [Ideas]
## Scope & Context

Currently, on objects we can only add txt/number/boolean and date field.

It woould be great to add a computed field that use the value of other fields to calculate it's own value. 
For instance a field calculating the estimated opportunity value following the number of employees on the company. 
Or, using the same use case as illustrated in  #2534, a field that count the numbe of opportunities or closed deal by referer.

A notion like approach could be inspiring :
![image](https://github.com/twentyhq/twenty/assets/9041351/a0076e87-5334-454d-8eac-aa9e843172af)

**Comment by FelixMalfait** [comment]
Yes that would be super cool 😍

**Comment by Bonapara** [comment]
Related suggestion: https://github.com/twentyhq/twenty/issues/7139

**Comment by thinkbig1979** [comment]
Sorry I missed this topic when adding the feature request.  I searched for a few terms, but somehow 'computed' was not one of them 😅.
Anyhow, this would be a great addition. 
Twenty is so nice and well thought out, but just missing a few powerful features that would give it the edge over pretty much everything else in this space. 
Keep up the great work! 💪

**Comment by eshut** [comment]
+1 here. It's the only thing stops me from using this CRM. You can see the open-source EspoCRM formulas (it's on php 🙄 and sometimes broken...), it allows to link different entities, e.g Opportunities + Accounts, so I can calculate all the related opportunity.amount to some Account and show the result in some field.

And also the missing thing here is to create different (custom) entities, where I can later also use some formulas. E.g. in data-models, after I create a new Objact(page) I expect it to have many-to-many or many-to-one, etc.. relationships to create some custom fields with links to different Objects @Bonapara 

![image](https://github.com/user-attachments/assets/04261cac-b13d-4305-92d0-7573de61a927)

**Reply by FelixMalfait** [reply]
@eshut Thanks for sharing! there are multiple ways to do computed fields. For "cached" computed fields that are re-computed when the source value changes (it seems that's what Espo does), we will release a major feature "Workflows" in early beta end of October which should allow hacking that kind of things. It will force you to call the API in a workflow step at first but as we iterate we will make it easier to build things like that without technical knowledge

**Reply by eshut** [reply]
@FelixMalfait Wow, that sounds great. I would be happy to test this out and share my feedback on it.

**Reply by timdmackey** [reply]
Is a workflow still the right way to accomplish this? Testing out twenty and I'm creating a setup for an annual school program. I have a many to one relation between annual programs (many) and schools (one), and want to show the most recent program in the school view.

**Reply by FelixMalfait** [reply]
@timdmackey yes unfortunately we aren't there yet. It's still 100% something we want to get done but it will take time.
We are now ready from an api/backend perspective but need to find the right UX to display those second-order fields (could be something at the data model level or the view level)

**Comment by BOHEUS** [comment]
Related to https://github.com/twentyhq/core-team-issues/issues/813

**Comment by redvaldhj** [comment]
+1 — running into this for a school admin use case (tracking which academic year(s) a person is linked to, via a custom object related to Person).

Without a computed Name field, every record shows as "Untitled" until someone manually types a label — an extra step and a source of inconsistent titles across staff. Tried the Workflows route, but the "record created" trigger explicitly warns against manually-created records (fires before the user finishes the form), and the Code step for updates seems undocumented right now.

Really impressed with Twenty otherwise — this is the one thing standing between us and adopting it.

**Reply by BOHEUS** [reply]
You can work around this for time being with Code node

**Comment by iamapay** [comment]
No news on this ? Any ETA ?

**Reply by Bonapara** [reply]
We started working on new record forms (with validations) i think it might help the formula field

**Reply by iamapay** [reply]
Good luck. This is such an important feature to us (and lot's of other users apparently), it's going to make Twenty CRM stellar.
